pkg/apis/core/v1alpha1/helper/helper.go

```diff
@@ -742,3 +742,37 @@ func GetExtensionResourceState(extensionsResourcesStates []gardencorev1alpha1.Ex
 	}
 	return -1, nil
 }
+
+func RemoveGardenerResourceData(gardenerResourceDataList []gardencorev1alpha1.GardenerResourceData, name string) []gardencorev1alpha1.GardenerResourceData {
```

Comment:
Same as with the ShootStateSync controller PR, if I do that I would have to add gardencorev1alpha1 to the v1beta1/helper/helper.go file since GardenerResourceData (part of the ShootState) is still only in v1alpha1